### PR TITLE
Add SignalProducer restart operator

### PR DESCRIPTION
**Pull request by elfenlaid**

### CODE REVIEW

File: ReactiveCocoa/Swift/SignalProducer.swift [hunk 1]
```diff
@@ -1416,6 +1416,41 @@ extension SignalProducerType {
...
+
+		return SignalProducer { observer, disposable in
+			let serialDisposable = SerialDisposable()
+			disposable.addDisposable(serialDisposable)
+
+			func innerRestart() {
+				self.startWithSignal { signal, signalDisposable in
+					serialDisposable.innerDisposable = signalDisposable
+
+					signal.observe { event in
```
Comment (filblue) [inline ReactiveCocoa/Swift/SignalProducer.swift]: you have to interrupt active signal when the restarting one is interrupted

Comment (elfenlaid) [inline ReactiveCocoa/Swift/SignalProducer.swift]: I've somewhat fixed it, though I'm not sure that I'm completely understand the problem

Comment (filblue) [inline ReactiveCocoa/Swift/SignalProducer.swift]: yep, that makes it safer. there was no leak here, still it never hurts to handle all the disposables you get.

File: ReactiveCocoa/Swift/SignalProducer.swift [hunk 2]
```diff
@@ -1416,6 +1416,47 @@ extension SignalProducerType {
...
+				self.startWithSignal { signal, signalDisposable in
+					let iterationDisposable = CompositeDisposable()
+					iterationDisposable.addDisposable(signalDisposable)
+
+					serialDisposable.innerDisposable = iterationDisposable
+
+					let activeSignalDisposable = signal.observe { event in
+						if case .Completed = event {
+							let date = scheduler.currentDate.dateByAddingTimeInterval(delay)
+							let schedulerDisposable = scheduler.scheduleAfter(date) {
```
Comment (filblue) [inline ReactiveCocoa/Swift/SignalProducer.swift]: actually there was a leak here, but you've handled it as well 👍

File: ReactiveCocoa/Swift/SignalProducer.swift (hunk 1 above)
Comment (elfenlaid) [inline ReactiveCocoa/Swift/SignalProducer.swift]: ah, thanks

**Pull request closed without merging**